add, change and delete email function

SerhioKAS opened the pull request:
Розкрив клас EmailContact. Створив перевірку на правильність вводу ел. пошти створив функції додавання/редагування/видалення ел. пошти

SerhioKAS committed:
add, change and delete email function

tymoshchuk-vladyslav commented on bot_assistant/classes.py:
```diff
@@ -145,6 +152,23 @@ def change_phone(self, new_phone):
+        """
+        if len(self.email) == 0:
+            return "контакт ще не має ел. пошти."
+        else:
```
тут додай по аналогії 139 стрічка якщо тільки 1 майл то його зразу і заміняти

tymoshchuk-vladyslav commented on bot_assistant/classes.py:
```diff
@@ -166,6 +190,17 @@ def delete_phone(self):
+        Метод для видалення ел. пошти у контакта.
+        """
+        for eml in self.email:
+            if eml.value == email:
```
== заміни на in

tymoshchuk-vladyslav commented on bot_assistant/main.py:
```diff
@@ -224,6 +259,19 @@ def delete_phone(args):
+@input_error
+def delete_email(data):
+    """
+    Функція для редагування електронної пошти контакту.
```
для видалення

tymoshchuk-vladyslav commented on bot_assistant/main.py:
```diff
@@ -512,7 +560,11 @@ def fun_name(fun):
         "delete_phone": delete_phone,
-        "show_contact": show_contact
+        "show_contact": show_contact,
+        'add_email': add_email,
```
редагуй скобки так як кругом

tymoshchuk-vladyslav reviewed (commented):
Додати в def parser(): інфу по всіх командах окремою стрічками по емейлах, а також в функцію def help().......

SerhioKAS committed:
defects changed

Додав email у def parcer та def helps, підкорегував def change_email, записи в словнику

TimVladislav13010 commented:
падає дає помилку...
Введіть будь ласка команду: (або використай команду help)
add vlad [+380687410000](tel:+380687410000)
Vlad was added with [+380687410000](tel:+380687410000)
Введіть будь ласка команду: (або використай команду help)
show contact
  Name:Vlad
Phones:['[+380687410000](tel:+380687410000)']
Address:[]
Bday:None
Email:[]
------------------------------
Введіть будь ласка команду: (або використай команду help)
add email vlad
'list' object has no attribute 'title'
Введіть будь ласка команду: (або використай команду help)
add email vlad
'list' object has no attribute 'title'
Введіть будь ласка команду: (або використай команду help)
add email
'list' object has no attribute 'title'
Введіть будь ласка команду: (або використай команду help)

і дивись між методами 1 пробіл між класами 2 пробіли між функціями 2 пробіли) бо світиться все)

517 і 527 стрічка 2 рази .replace("add email", "add_email") прописаний

в одну стрічку запиши ці парсери з мейлами.... (отредактировано)

SerhioKAS committed:
fixed errors

now programm can add/change/delete emails.

tymoshchuk-vladyslav reviewed (commented):
видали пайкеш з коміта

SerhioKAS committed:
free space deleted

tymoshchuk-vladyslav reviewed (commented):
знову пайкеш)

tymoshchuk-vladyslav committed:
edit add email and change email

tymoshchuk-vladyslav committed:
edit delete email

tymoshchuk-vladyslav committed:
Merge branch 'dev' into 18-19-20/add_email_change_email_delete_email
